src/tests/Interop/NativeLibrary/MainProgramHandle/MainProgramHandleTests.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+using Xunit;
+
+public static class MainProgramHandleTests
+{
+    private static IntPtr s_handle;
+
+    static MainProgramHandleTests() => NativeLibrary.SetDllImportResolver(typeof(MainProgramHandleTests).Assembly,
+        (string libraryName, Assembly asm, DllImportSearchPath? dllImportSearchPath) =>
+        {
+            if (libraryName == "Self")
+            {
+                s_handle = NativeLibrary.GetMainProgramHandle();
+                Assert.NotEqual(IntPtr.Zero, s_handle);
+                return s_handle;
+            }
+
+            return IntPtr.Zero;
+        });
+
+    public static int Main()
+    {
+        try
+        {
+            free(s_handle);
```

Comment:
Why do we free `s_handle` here? That seems like we're asking for trouble.  `free (IntPtr.Zero)` would also work (`free (NULL)` is a no-op call - so we would just test that the we got the right symbol, which is what we want).

I'm assuming the failure on the mono interpreter legs is somehow related (we need to keep the native handle around longer for some reason?)

Reply:
Makes sense, thanks!

Reply:
I'm trying with `getppid()`. Seems like it will work everywhere (test on osx-arm64 and android-arm64). Lets try that after this CI run completes.

Reply:
Nvm, it didn't help. IL interp needs to be fixed. :)